src/librbd/image/OpenRequest.cc

```diff
@@ -614,6 +615,23 @@ Context *OpenRequest<I>::handle_set_snap(int *result) {
     return nullptr;
   }
 
+  return finalize(*result);
+}
+
+template <typename I>
+Context *OpenRequest<I>::finalize(int r) {
+  if (r == 0) {
+    auto io_scheduler_cfg =
+      m_image_ctx->config.template get_val<std::string>("rbd_io_scheduler");
+
+    if (io_scheduler_cfg == "simple" && m_image_ctx->snap_id == CEPH_NOSNAP &&
```

Comment:
Nit: if we don't modify the set snap state machine to create/destroy the IO scheduler when switching between a snapshot and HEAD, perhaps we can just leave off the ```m_image_ctx->snap_id == CEPH_NOSNAP``` check?